src/app/components/Settings/NodeSettings.tsx

```diff
@@ -0,0 +1,209 @@
+// Huge bastardization of ant design's Transfer component
+import React from 'react';
+import { connect } from 'react-redux';
+import { Form, Input, Button, Icon, Drawer, message } from 'antd';
+import { editNodeField, updateNodeUrl, updateMacaroons } from 'modules/node/actions';
+import { changePassword, setPassword, cancelChangePassword } from 'modules/crypto/actions';
+import CreatePassword from 'components/CreatePassword';
+import InputAddress from 'components/SelectNode/InputAddress';
+import UploadMacaroons from 'components/SelectNode/UploadMacaroons';
+import { AppState } from 'store/reducers';
+
+interface StateProps {
+  url: AppState['node']['url'];
+  readonlyMacaroon: AppState['node']['readonlyMacaroon'];
+  adminMacaroon: AppState['node']['adminMacaroon'];
+  isNodeChecked: AppState['node']['isNodeChecked'];
+  editingNodeField: AppState['node']['editingNodeField'];
+  isUpdatingNodeUrl: AppState['node']['isUpdatingNodeUrl'];
+  updateNodeUrlError: AppState['node']['updateNodeUrlError'];
+  isUpdatingMacaroons: AppState['node']['isUpdatingMacaroons'];
+  updateMacaroonsError: AppState['node']['updateMacaroonsError'];
+  isChangingPassword: AppState['crypto']['isChangingPassword'];
+  password: AppState['crypto']['password'];
+}
+
+interface DispatchProps {
+  editNodeField: typeof editNodeField;
+  updateNodeUrl: typeof updateNodeUrl;
+  updateMacaroons: typeof updateMacaroons;
+  changePassword: typeof changePassword;
+  cancelChangePassword: typeof cancelChangePassword;
+  setPassword: typeof setPassword;
+}
+
+type Props = StateProps & DispatchProps;
+
+class NodeSettings extends React.Component<Props> {
+
+  componentWillUpdate(nextProps: Props) {
+    const { 
+      isNodeChecked, 
+      isChangingPassword, 
+      password, 
+      isUpdatingMacaroons, 
+      adminMacaroon, 
+      readonlyMacaroon 
+    } = this.props;
+
+    if (isNodeChecked !== nextProps.isNodeChecked && nextProps.isNodeChecked) {
+          // isNodeChecked false -> true
+          message.success(`Connected to ${nextProps.url}`, 2);
+    }
+    if (isChangingPassword !== nextProps.isChangingPassword && !isChangingPassword
+        && password !== nextProps.password) {
+          // isChangingPassword true -> false and password changed
+          message.success('Password Updated', 2);
```

Comment:
This could be moved to a method that's passed for CreatePassword's onCreatePassword, since that action can't fail, so you might as well call them at the same time.

This also helps you get rid of `isChangingPassword` from redux state.

Reply:
I'm not 100% sure I am following you here. Are you saying to create a private function in `NodeSettings`  that is assigned to `<CreatePassword onCreatePassword={this.newFunction} ...` and in this function, call both `this.props.setPassword()` and `message.success()`?

Reply:
Yep, that's what I had in mind! Though after the change I proposed here: https://github.com/wbobeirne/joule-extension/pull/133#discussion_r247167131 you would want it in the same place as the drawer close, rather than the `onCreatePassword`, so it may have to remain in `componentWillUpdate` after all.